internal/backend/runtime/omni/controllers/omni/cluster_workload_proxy_status_test.go

```diff
@@ -218,11 +227,50 @@ func (m *mockWorkloadProxyReconciler) Reconcile(cluster resource.ID, aliasToUpst
 		m.data = map[resource.ID]map[string][]string{}
 	}
 
-	m.data[cluster] = aliasToUpstreamAddresses
+	newCluster := true
+
+	for als := range rd.AliasPort {
+		for clusterID, alsToHost := range m.data {
+			if _, ok := alsToHost[als]; ok && clusterID != cluster {
+				panic(fmt.Errorf("alias %q already exists and used by cluster %q", als, clusterID))
```

Comment:
I don't quite see panic here doing anything good for real?

Reply:
This should't happen actually, so it's just a check in tests to ensure no bugs.